### PR TITLE
[feat] support pe ltr in train wrapper

### PR DESCRIPTION
**Pull request by chengaofei**
Feature/support pe ltr in train wrapper

### CODE REVIEW

File: tzrec/modules/rebuild_loss.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+from typing import Dict
+
+import numpy as np
+import torch
+from scipy.optimize import minimize, nnls
+
+
+class ParetoDynamicLossWeight(torch.nn.Module):
+    """Pareto front dynamic loss weight get losses for multi target."""
```
Comment (tiankongdeguiji) [inline tzrec/modules/rebuild_loss.py]: Dynamic loss weights based on the Pareto-Efficient for multi-task learning.

File: tzrec/models/multi_task_rank.py [hunk 2]
```diff
@@ -84,12 +89,21 @@ def init_loss(self) -> None:
                     reduction=reduction,
                     suffix=f"_{tower_name}",
                 )
+                if self._use_pareto_loss_weight:
+                    self._pareto_init_weight_cs.append(
+                        task_tower_cfg.pareto_min_loss_weight
+                    )
+
+        if self._use_pareto_loss_weight:
```
Comment (tiankongdeguiji) [inline tzrec/models/multi_task_rank.py]: move it into ParetoDynamicLossWeight __init__

File: tzrec/loss/pe_mtl_loss.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2024, Alibaba Group;
```
Comment (tiankongdeguiji) [inline tzrec/loss/pe_mtl_loss.py:1]: rebuild_loss.py -> loss/pe_mtl_loss.py

File: tzrec/modules/rebuild_loss.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+# limitations under the License.
+
+from typing import Dict
+
+import numpy as np
+import torch
+from scipy.optimize import minimize, nnls
+
+
+class ParetoDynamicLossWeight(torch.nn.Module):
```
Comment (tiankongdeguiji) [inline tzrec/modules/rebuild_loss.py]: ParetoEfficientMultiTaskLoss

File: docs/source/models/loss.md [hunk 5]
```diff
@@ -99,3 +99,38 @@ DISTRIBUTE BY user_id
...
+    feature_groups {
+        group_name: "all"
+        feature_names: "user_id"
+        ...
+        group_type: DEEP
+    }
+    ${model_name} {
+        task_towers {
+            ...
+            ${loss_name} {}
```
Comment (tiankongdeguiji) [inline docs/source/models/loss.md]: losses {
                ${loss_name} {}
            }

File: docs/source/models/loss.md [hunk 6]
```diff
@@ -99,3 +99,38 @@ DISTRIBUTE BY user_id
 SORT BY user_id asc,time_stamp asc
 ;
 ```
+
+## pe_mtl_loss
+
+该方法适用于多任务场景，主要作用于损失权重的学习。需要结合别的损失函数一起使用，根据每次batch中的数据的每个分量损失函数，给与每个分量损失不同的权重。https://dl.acm.org/doi/10.1145/3298689.3346998
```
Comment (tiankongdeguiji) [inline docs/source/models/loss.md]: Pareto-Efficient Algorithm for Multiple Objective Optimization，该方法适用于多任务场景，自动根据帕累托最优最优的原则调整不同目标的损失权重。

**Review by tiankongdeguiji — APPROVED** [review]
